packages/integration-tests/src/test-projects/conversation-handler/amplify/custom_handler.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  ConversationTurnEvent,
+  ExecutableTool,
+  handleConversationTurnEvent,
+} from '@aws-amplify/ai-constructs/conversation/runtime';
+import { ToolResultContentBlock } from '@aws-sdk/client-bedrock-runtime';
+
+const thermometer: ExecutableTool = {
+  name: 'thermometer',
+  description: 'Returns current temperature in Seattle',
+  execute: (): Promise<ToolResultContentBlock> => {
+    return Promise.resolve({
+      // We use this value in test assertion.
+      text: '75F',
```

Comment:
does this need to be the same shape as the defined schema?

Reply:
nvm i see we aren't using custom handler in that schema

Reply:
> does this need to be the same shape as the defined schema?

No, the input schema is something we give to LLM to let it know what is the shape of json that tool expects. (we do not test that particular aspect well, but I expect to make this up later when these components mature more).
The output from tool can be anything really. LLM is handling it internally somehow.